### ISSUE TITLE
Different bandwidth limits for two transfer directions

### ISSUE THREAD

**Issue by Shawnxm** [esnet/iperf#1093]
# Context

* Version of iperf3: iperf 3.9+ (cJSON 1.7.13)

* Hardware: Desktop

* Operating system (and distribution, if any): Linux 16.04

# Enhancement Request

* Current behavior: iPerf3 currently supports concurrent bidirectional transfer (`--bidir`).

* Desired behavior: Can we set different bandwidth limits for each direction?

* Implementation notes: `./iperf3 -c [server ip] -p [server port] -u -b 10M --bidir` will lead to 10Mbps limit for both directions. It would be great if some options like -bu (uplink, or send) -ud (downlink, receive) can be supported.

**Comment by lealog** [comment]
This is a great feature to have on iperf3 👍

**Comment by bmah888** [comment]
This is a valid use case, although one could simulate this now by running two independent iperf3 tests between the hosts in question, one in each direction.

**Comment by davidBar-On** [comment]
Submitted PR #1102 with a suggested enhancement - adding optional argument to `--bidir` that specifies the distinct server's bitrate.

**Comment by potpara** [comment]
Apologies for cutting in, is this option already implemented in iperf3 3.11? I'm trying to set different uplink and downlink bit rate for bidirectional test but seems it doesn't work.

**Comment by bmah888** [comment]
> Apologies for cutting in, is this option already implemented in iperf3 3.11? I'm trying to set different uplink and downlink bit rate for bidirectional test but seems it doesn't work.

No, not yet. If the pull request is still open, it hasn't been merged to the mainline. Releases are selected points on the mainline.

**Comment by potpara** [comment]
Thanks for quick reply.
